packages/core/src/app/shipping/SingleShippingForm.tsx

```diff
@@ -145,8 +145,22 @@ class SingleShippingForm extends PureComponent<
             shippingAddress,
             isBillingSameAsShipping,
             customerMessage,
+            values,
+            setFieldValue,
         } = this.props;
 
+        const stateOrProvinceCodeFormField = getFields(values && values.shippingAddress?.countryCode).find(
+            ({ name }) => name === 'stateOrProvinceCode',
+        );
+
+        if (
+            stateOrProvinceCodeFormField &&
+            shippingAddress?.stateOrProvinceCode &&
+            values.shippingAddress?.stateOrProvinceCode !== shippingAddress.stateOrProvinceCode
+        ) {
+            setFieldValue('shippingAddress.stateOrProvinceCode', shippingAddress.stateOrProvinceCode);
+        }
+
```

Comment:
An interesting formik v2 bug during manual testing.  It wasn’t an issue previously.

Initially, the `stateOrProvinceCode` field is not present when the shipping step is first shown. However, after some time, it appears in the form with an empty value if the countries does have `stateOrProvinceCode` options.

To address this, manually setting `stateOrProvinceCode` in formik v2.

Reply:
Maybe add some comment on this as to why we need to have this logic in place.